### ISSUE TITLE
🐛 Bug: no-redundant-files conflicts with  n/no-unpublished-bin

### ISSUE THREAD

**Issue by cylewaitforit** [JoshuaKGoldberg/eslint-plugin-package-json#1246]
### Bug Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

When upgrading to 0.56.0 in which [no-redundant-files](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/blob/main/docs/rules/no-redundant-files.md) is a part of the recommended config the no-redundant-files rule flagged `"bin/index.js",` in the package.json as redundant. Removing it should result in lint passing.

### Actual

After removing `"bin/index.js",` in the package.json files, the [no-unpublished-bin](https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-unpublished-bin.md) rule from eslint-plugin-n throws 

```sh
error  npm ignores 'bin/index.js'. Check 'files' field of 'package.json' or '.npmignore'  n/no-unpublished-bin
```

### Version

0.56.0

### ESLint Version

9.34.0

### Additional Info

"eslint-plugin-n": "17.21.0"

The [no-unpublished-bin documentation](https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-unpublished-bin.md) states:

>If `npm` ignores the files in `bin` field, this rule warns the files.
>
>- If `files` field does not includes the files in `bin` field.

I don't know which rule is correct. Filing the issue here first because enabling this rule is how the discrepancy was found.

You can recreate this in https://github.com/JoshuaKGoldberg/OctoGuide/pull/260.

**Comment by michaelfaith** [comment]
I see there's activity in the issue you updated for the other plugin.  I'll go ahead and close this one on our side.
